Source/Core/Core/PowerPC/PPCSymbolDB.cpp

```diff
@@ -296,14 +291,37 @@ bool PPCSymbolDB::LoadMap(const std::string& filename, bool bad)
     if (section_name.empty())
       continue;
 
-    // Detect two columns with three columns fallback
+    // Column detection heuristic
     if (column_count == 0)
     {
-      const std::string_view stripped_line = StripSpaces(line);
-      if (std::count(stripped_line.begin(), stripped_line.end(), ' ') == 1)
-        column_count = 2;
-      else
+      constexpr auto is_hex_str = [](const std::string& s) {
+        return !s.empty() && s.find_first_not_of("0123456789abcdefABCDEF") == std::string::npos;
+      };
+      const std::string stripped_line(StripSpaces(line));
+      std::istringstream oss(stripped_line);
+      std::string word;
+
+      // Two columns format:
+      // 80004000 zz_80004000_
+      if (!(oss >> word) || word.length() != 8 || !is_hex_str(word))
+        continue;
+      column_count = 2;
+
+      // Three columns format:
+      //  Starting        Virtual
+      //  address  Size   address
+      //  -----------------------
+      if (oss && oss >> word && is_hex_str(word) && oss >> word && is_hex_str(word))
```

Comment:
Sorry if a dumb question but I want to check my understanding.

The first word has been read, that's a hex value.  If the second value is read here, it's a size, and not a hex string.  The third value is a hex string.  Shouldn't the first `is_hex_str` in this check be a `!is_hex_str` ?

Reply:
All columns are hexstring: address, size and vaddrress.

Reply:
Ohhh.  Thank you for the clarification